include/nbl/core/alloc/PoolAddressAllocator.h

```diff
@@ -47,9 +47,7 @@ class PoolAddressAllocator : public AddressAllocatorBase<PoolAddressAllocator<_s
 
         static constexpr bool supportsNullBuffer = true;
 
-        #define DUMMY_DEFAULT_CONSTRUCTOR PoolAddressAllocator() : blockSize(1u), blockCount(0u) {}
-        GCC_CONSTRUCTOR_INHERITANCE_BUG_WORKAROUND(DUMMY_DEFAULT_CONSTRUCTOR)
-        #undef DUMMY_DEFAULT_CONSTRUCTOR
```

Comment:
this sort of fix was better, because its an actual GCC bug

Reply:
it was my idea to get rid of it. I dont get this fix, what it does is defining default ctor but only in gcc versions below 7.2 (and in fact msvc as well because the `#if` is poorly written XD -- thats why msvc build even worked), while default ctor is expected to be there always, regardless of platform/compiler/config

Reply:
ah ok.. so it turns out every compiler needs it?

Reply:
yeah, tbh i dont know what this fix is for

Reply:
it seemed that MSVC can do without the default ctor

Reply:
nah, it's just that `#if` was poorly written and it was actually creating this ctor when `__GNUC__` was not defined as well

Reply:
ok